activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -483,7 +483,7 @@ def log(sql, name = "SQL", binds = [], statement_name = nil)
 
       def translate_exception(exception, message)
         # override in derived class
-        ActiveRecord::StatementInvalid.new(message, exception)
+        ActiveRecord::StatementInvalid.new(message)
```

Comment:
It actually doesn't. As long as `#translate_exception` method is called from within `rescue ... end`, it'll capture `$!` automatically.

```ruby
def raise_again
  raise SecondError
end

begin
  raise FirstError
rescue
  raise SecondError # This'll capture the first error.
  raise_again       # This'll also capture the first error.
end

raise SecondError # This won't.
```

I've added several assertions just in case `#translate_exception` is moved outside of `rescue ... end`.